tests/validator_test.go

```diff
@@ -329,4 +337,594 @@ var _ = Describe("Template validator", func() {
+				vm = NewVirtualMachine(vmi, false)
+				vm.ObjectMeta.Annotations = map[string]string{
+					"vm.kubevirt.io/template":           "nonexisting-vm-template",
+					"vm.kubevirt.io/template-namespace": strategy.GetTemplatesNamespace(),
+					"vm.kubevirt.io/validations": `[
+												 {
+														"name": "LimitCores",
+														"path": "jsonpath::.spec.domain.cpu.cores",
+														"message": "Core amount not within range",
+														"rule": "integer",
+														"min": 1,
+														"max": 4
+        										 }
+												]`,
+				}
+				Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Failed to create VM")
+			})
+			It("[test_id:5046]: should fail to create a VM based on the VM validation rules", func() {
+				vmi = addDomainResourcesToVMI(vmi, 5, "q35", "64M")
+				vm = NewVirtualMachine(vmi, false)
+				vm.ObjectMeta.Annotations = map[string]string{
+					"vm.kubevirt.io/template":           "nonexisting-vm-template",
+					"vm.kubevirt.io/template-namespace": strategy.GetTemplatesNamespace(),
+					"vm.kubevirt.io/validations": `[
+												 {
+														"name": "LimitCores",
+														"path": "jsonpath::.spec.domain.cpu.cores",
+														"message": "Core amount not within range",
+														"rule": "integer",
+														"min": 1,
+														"max": 4
+        										 }
+												]`,
+				}
+				Expect(apiClient.Create(ctx, vm)).To(HaveOccurred(), "Failed to create VM")
+			})
+			It("[test_id:5047]: should fail to create a VM based on the VM validation rules", func() {
+				vmi = addDomainResourcesToVMI(vmi, 5, "q35", "64M")
+				vm = NewVirtualMachine(vmi, false)
+				vm.ObjectMeta.Annotations = map[string]string{
+					"vm.kubevirt.io/validations": `[
+												 {
+														"name": "LimitCores",
+														"path": "jsonpath::.spec.domain.cpu.cores",
+														"message": "Core amount not within range",
+														"rule": "integer",
+														"min": 1,
+														"max": 4
+        										 }
+												]`,
+				}
+				Expect(apiClient.Create(ctx, vm)).To(HaveOccurred(), "Failed to create VM")
+			})
+			It("[test_id:5175]: VM with validations without template", func() {
+				vmi = addDomainResourcesToVMI(vmi, 3, "q35", "64M")
+				vm = NewVirtualMachine(vmi, false)
+				vm.ObjectMeta.Annotations = map[string]string{
+					"vm.kubevirt.io/validations": `[
+												 {
+														"name": "LimitCores",
+														"path": "jsonpath::.spec.domain.cpu.cores",
+														"message": "Core amount not within range",
+														"rule": "integer",
+														"min": 1,
+														"max": 4
+        										 }
+												]`,
+				}
+				Expect(apiClient.Create(ctx, vm)).ToNot(HaveOccurred(), "Failed to create VM")
+			})
+		})
+	})
+
+	Context("Certificates", func() {
+		var (
+			secretName = "virt-template-validator-certs"
+		)
+		It("[test_id:4375] Test refreshing of certificates", func() {
```

Comment:
This test is not the same as the one in `kubevirt-template-validator` repository. This one does not actually test certificate rotation inside the template validator.

The original test may be more complicated to port into golang. If you don't want to do it in this PR, then please mark this test as pending, write a TODO comment, and open a new issue, so we don't forget about it.